File: rb/features.rb
# BreakingBadQuotes SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module BreakingBadQuotesFeatures
  def self.make_feature(name)
    case name
    when "base"
      BreakingBadQuotesBaseFeature.new
    when "test"
      BreakingBadQuotesTestFeature.new
    else
      BreakingBadQuotesBaseFeature.new
    end
  end
end

File: py/features.py
# BreakingBadQuotes SDK feature factory

from feature.base_feature import BreakingBadQuotesBaseFeature
from feature.test_feature import BreakingBadQuotesTestFeature


def _make_feature(name):
    features = {
        "base": lambda: BreakingBadQuotesBaseFeature(),
        "test": lambda: BreakingBadQuotesTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
